selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java

```diff
@@ -860,50 +860,49 @@ public void scrollToItemByPath(String path) {
   }
 
   /**
-   * Expands specified path in project explorer Path given in format like like below:
+   * Expands specified {@code path} in project explorer. Path given in format like below:
    * TestProject/src/main/java/org.eclipse.che.examples
    *
    * @param path path for expand
    */
   public void expandPathInProjectExplorer(String path) {
-    String[] pathItems = path.split("/");
-    String currentItem = pathItems[0].replace('.', '/');
-    String currentPath = currentItem;
-
-    waitItem(currentPath);
-    openItemByPath(currentPath);
-    for (int i = 1; i < pathItems.length; i++) {
-      currentItem = pathItems[i].replace('.', '/');
-      currentPath += "/" + currentItem;
-      waitItem(currentPath);
-      loader.waitOnClosed();
-      openItemByPath(currentPath);
+    List<String> itemsPaths = splitPathOnSubitems(path);
+    ListIterator<String> iterator = itemsPaths.listIterator();
+
+    while (iterator.hasNext()) {
+      String currentItem = iterator.next();
+      if (!iterator.hasNext()) {
+        waitItem(currentItem);
```

Comment:
Code inside `if` looks useless so as we are opening this item as next one at the previous iteration.

Reply:
Without this `if` we will have `IndexOutOfBounds` exception in all cases

Reply:
I see.
What about simplifying method a little bit:
```java
    for (int i = 0; i < itemsPaths.size(); i++) {
      String currentItem = itemsPaths.get(i);
      int nextItemIndex = i + 1;
      
      // don't open if next item has already been visible
      if (nextItemIndex < itemsPaths.size()
          && seleniumWebDriverHelper.isVisible(By.xpath(format(PROJECT_EXPORER_ITEM_TEMPLATE, itemsPaths.get(nextItemIndex))))) {
        continue;
      }

      waitItem(currentItem);
      openItemByPath(currentItem);
    }
```